Typescript test failing due to compilation error

## Expected Behavior
Upon a fresh git clone, I should be able to run the tests successfully. 

## Actual Behavior
The typescript test seems to have a compilation error. I've included the output below.

```
...
$ node zx.mjs tests/typescript.ts
(node:68856) UnhandledPromiseRejectionWarning: Error: Cannot find module '/Users/ivraj/work/projects/zx/dist/index.cjs'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:321:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:534:18)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:888:27)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/Users/ivraj/work/projects/zx/tests/typescript.cjs:17:13)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:68856) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:68856) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
internal/process/esm_loader.js:74
    internalBinding('errors').triggerUncaughtException(
                              ^

Error: Cannot find module '/Users/ivraj/work/projects/zx/dist/index.cjs'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:321:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:534:18)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:888:27)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/Users/ivraj/work/projects/zx/tests/typescript.cjs:17:13)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32) {
  code: 'MODULE_NOT_FOUND',
  path: '/Users/ivraj/work/projects/zx/package.json',
  requestPath: '..'
}
Error: (node:68856) UnhandledPromiseRejectionWarning: Error: Cannot find module '/Users/ivraj/work/projects/zx/dist/index.cjs'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:321:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:534:18)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:888:27)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/Users/ivraj/work/projects/zx/tests/typescript.cjs:17:13)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:68856) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:68856) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
internal/process/esm_loader.js:74
    internalBinding('errors').triggerUncaughtException(
                              ^

Error: Cannot find module '/Users/ivraj/work/projects/zx/dist/index.cjs'. Please verify that the package.json has a valid "main" entry
    at tryPackage (internal/modules/cjs/loader.js:321:19)
    at Function.Module._findPath (internal/modules/cjs/loader.js:534:18)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:888:27)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/Users/ivraj/work/projects/zx/tests/typescript.cjs:17:13)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32) {
  code: 'MODULE_NOT_FOUND',
  path: '/Users/ivraj/work/projects/zx/package.json',
  requestPath: '..'
}
    at file:///Users/ivraj/work/projects/zx/test.mjs:96:25
    exit code: 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zx@4.3.0 test: `node zx.mjs test.mjs`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zx@4.3.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/ivraj/.npm/_logs/2022-01-17T21_04_36_570Z-debug.log
```

## Steps to Reproduce the Problem

1. Clone the project (`git clone https://github.com/google/zx.git`)
2. `cd` into the directory and install the deps. (`cd zx && npm i`)
3. Run tests. (`npm run test`)

Most of the tests should run fine, but once you get to the Typescript test (`test.mjs:95`), it should fail with the above error. 

## Specifications

- Version: 4.3.0
- Platform: macOS Big Sur 11.2.3 (Apple M1), Node 14.18.3

I tried debugging the issue on my own for a bit. I found it curious that this doesn't pop up in CI. I'm wondering whether this has ever happened to you locally @antonmedv?

--- a/package.json
+++ b/package.json
@@ -21,6 +21,7 @@
     "node": ">= 14.13.1"
   },
   "scripts": {
+    "pretest": "npm run build",
     "test": "node zx.mjs test.mjs",
     "build": "node zx.mjs .github/scripts/build.mjs"
   },

